src/components/MessageInput/AttachmentPreviewList.tsx

```diff
@@ -28,6 +65,113 @@ export const AttachmentPreviewList = () => {
   );
 };
 
+type AttachmentPreviewProps<A extends LocalAttachment = LocalAttachment> = {
+  attachment: A;
+  removeAttachment: (id: string) => void;
+  handleRetry?: (attachment: A) => void | Promise<void>;
+};
+
+const VoiceRecordingPreview = ({
+  attachment,
+  handleRetry,
+  removeAttachment,
+}: AttachmentPreviewProps<VoiceRecordingAttachment>) => {
+  const { audioRef, isPlaying, secondsElapsed, togglePlay } = useAudioController();
+
+  return (
+    <div
+      className='str-chat__attachment-preview-voice-recording'
+      data-testid='attachment-preview-voice-recording'
+    >
+      <audio ref={audioRef}>
+        <source data-testid='audio-source' src={attachment.asset_url} type={attachment.mime_type} />
+      </audio>
+      <PlayButton isPlaying={isPlaying} onClick={togglePlay} />
+
+      <button
+        className='str-chat__attachment-preview-delete'
+        data-testid='file-preview-item-delete-button'
+        disabled={attachment.$internal?.uploadState === AttachmentUploadState.UPLOADING}
```

Comment:
Can you explain this `$internal` stuff? Is it something voice message related? I haven't seen it for attachments before.

Reply:
This are internal metadata that are not shared with the back-end. I decided to keep them "isolated" under this key and just remove them before sending the message with the attachment to the back-end. Current implementation with image and file uploads seems to me clunky in that it does not isolate these locally important metadata and even use different names for those attributes, that are later removed before the submission. 

I would like to propose this approach also to other types of attachments gradually to make our code more readable.

Reply:
I see, here's what worries me: integrators implementing custom attachment previews shouldn't care about what is and what isn't part of our backend schema. But in this case they will have to use this `$internal` property, and they may find it a bit weird.